paddle/fluid/operators/detail/grpc_client.cc

```diff
@@ -196,9 +197,14 @@ bool RPCClient::Wait() {
   const size_t kReqCnt = req_count_;
   bool a[kReqCnt];
   std::vector<std::future<void>> waits(req_count_);
+  std::mutex mu;
 
   for (int i = 0; i < req_count_; i++) {
-    waits[i] = framework::AsyncIO([i, &a, this] { a[i] = Proceed(); });
+    waits[i] = framework::AsyncIO([i, &a, &mu, this] {
+      bool ret = Proceed();
+      std::lock_guard<std::mutex> l(mu);
```

Comment:
Do we need mutex?

Reply:
yes, this one is needed. I added it because I hit error. a[i] is set by different thread, which might not be visible by anther thread if not synced by lock.